india_compliance/gst_india/doctype/purchase_reconciliation_tool/purchase_reconciliation_tool.js

```diff
@@ -1179,21 +1180,61 @@ class ImportDialog {
     }
 
     async download_gstr(only_missing = true, otp = null) {
-        let method;
-        const args = { date_range: this.date_range, otp };
-        if (this.return_type === ReturnType.GSTR2A) {
-            method = "download_gstr_2a";
-            args.force = !only_missing;
-        } else {
-            method = "download_gstr_2b";
-        }
+        const authenticated_company_gstins = await this.authenticate_company_gstins();
+
+        const args = {
+            return_type: this.return_type,
+            company_gstin_list: authenticated_company_gstins,
+            date_range: this.date_range,
+            force: !only_missing,
+            otp
+        };
 
         this.frm.events.show_progress(this.frm, "download");
-        const { message } = await this.frm.call(method, args);
-        if (message && ["otp_requested", "invalid_otp"].includes(message.error_type)) {
-            const otp = await india_compliance.get_gstin_otp(message.error_type, this.frm.doc.company_gstin);
-            if (otp) this.download_gstr(only_missing, otp);
-            return;
+        await this.frm.call("download_gstr", args);
+    }
+
+    async authenticate_company_gstins() {
+        const { message: gstin_authentication_status } = await frappe.call({
+            method: "india_compliance.gst_india.utils.gstr.gstr.validate_company_gstins",
+            args: { company: this.frm.doc.company, company_gstin: this.company_gstin == "All" ? null : this.company_gstin },
+        });
+
+        if (!Object.keys(gstin_authentication_status).length) {
+            frappe.throw(__("Please set GSTIN credentials in GST Settings"));
+        }
+
+        for (let gstin of Object.keys(gstin_authentication_status)) {
+            if (gstin_authentication_status[gstin]) continue;
```

Comment:
How about just returning unauthenticated GSTINs from the backend if authenticated GSTINs are not useful?